src/ReefModEngine.jl

```diff
@@ -23,6 +23,17 @@ macro RME(func)
     end
 end
 
+"""
+Only for use when RME functions return non-error numeric results. 
+
+For example any getter retrieving a numberic property.
+
+ivOutplantCountPerM2(::Cstring)::Cdouble.
+"""
+macro getRME(func)
+    return esc(Meta.parse("@ccall RME.$(func)"))
+end
```

Comment:
Wow, how does this work? I thought you needed to explicitly specify what the return type was for `@ccall`?

Reply:
Unfortunately, we still need to specify the return type. 
```julia
count::Float64 = @getRME ivOutplantCountPerM2("<name>"::Cstring)::Cdouble
```
Most of the calls to the RME use an out variable and the return value is the integer returned. 
This error integer is checked in the `@RME` macro.
```julia
!(typeof(err) <: Cstring) && err != 0 && !(typeof(err) <: Cvoid)
```
For functions that return numeric types the condition `err != 0` causes an exception when nothing has gone wrong. 

The documentation is somewhat unclear about how these functions 'error' when the return value is not an error indicator.

Reply:
Oh I get it, the return type is part of the `func` being parsed.

Thanks!

Reply:
> For functions that return numeric types the condition err != 0 causes an exception when nothing has gone wrong.

Oh, I forgot about that. Yes it's a pain that 0 stands for both "an error occurred" and "no error occurred" 😢